### ISSUE TITLE
Gracefully handle lack of special environment to edit

### ISSUE THREAD

**Issue by mohkale** [xenodium/company-org-block#2]
Some code blocks (such as quote) give an error when you try to edit them, however with `company-org-block-edit-style` set to `auto`, `company-org-block` will just let the error propagate. 

This diff should gracefully handle it.

```diff
--- /home/mohkale/.config/dotfiles/prog/editors/emacs/lisp/straight/repos/company-org-block/company-org-block.el
+++ #<buffer company-org-block.el>
@@ -110,11 +110,16 @@
   (insert (make-string org-edit-src-content-indentation ?\s))
   (save-excursion
     (insert (format "\n#+end_%s" end)))
-  (cond ((eq company-org-block-edit-style 'auto)
-         (org-edit-special))
-        ((and (eq company-org-block-edit-style 'prompt)
-              (yes-or-no-p "Edit now?"))
-         (org-edit-special))))
+  (condition-case err
+      (cond ((eq company-org-block-edit-style 'auto)
+             (org-edit-special))
+            ((and (eq company-org-block-edit-style 'prompt)
+                  (yes-or-no-p "Edit now?"))
+             (org-edit-special)))
+    (user-error
+     (unless (string-equal "No special environment to edit here"
+                           (error-message-string err))
+       (signal (car err) (cdr err))))))
 
 (defun company-org-block--grab-symbol-cons ()
   "Return cons with symbol and t whenever prefix of < is found.
```

**Comment by xenodium** [comment]
Ah sorry, meant to reply before submitting fix in 027d0f0. Does this fix work for you?

**Comment by mohkale** [comment]
@xenodium 

Yep. Seems to be working :smile:.
